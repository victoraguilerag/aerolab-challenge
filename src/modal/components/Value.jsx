import React from 'react'
import PropTypes from 'prop-types'
import coin from '../../icons/coin.svg'

export function Value (props) {
  const { value, active, handleValueSelected } = props
  if (active) {
    return (
      <div className="active-value" style={styles.activeValue}>
        <div className="active-value-text" style={styles.activeValueText} onClick={(e) => handleValueSelected(value)}>
          {value}
          <img className="coin" style={styles.coin} width="24px" height="24px" src={coin} alt="coin" />
        </div>
      </div>
    )
  }
  return (
    <div className="value" style={styles.value} onClick={() => handleValueSelected(value)}>
      <div className="valueText" style={styles.valueText}>
        {value}
        <img className="coin" style={styles.coin} width="24px" height="24px" src={coin} alt="coin" />
      </div>
    </div>
  )
}

const styles = {
  'value': {
    'background': '#ededed',
    'borderRadius': '100px',
    'height': '40px',
    'margin': '0 10px',
    'cursor': 'pointer'
  },
  'valueText': {
    'fontSize': '1em',
    'color': '#a3a3a3',
    'height': '100%',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'textAlign': 'center',
    'margin': '0 10px'
  },
  'activeValue': {
    'background': '#0ad4fa',
    'borderRadius': '100px',
    'height': '40px',
    'margin': '0 5px',
    'cursor': 'pointer'
  },
  'activeValueText': {
    'color': '#ffffff',
    'fontSize': '1em',
    'height': '100%',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'textAlign': 'center',
    'margin': '0 10px'
  },
  'coin': {
    'margin': '0 3px'
  }
}

Value.propTypes = {
  value: PropTypes.number.isRequired,
  active: PropTypes.bool.isRequired,
  handleValueSelected: PropTypes.func.isRequired
}

export default Value
