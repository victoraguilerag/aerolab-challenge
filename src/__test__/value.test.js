import React from 'react'
import { Value } from '../modal/components/Value.jsx'
import renderer from 'react-test-renderer'
import userStub from '../stub/user.js'

const functionMock = () => console.log('function mock');

test('Value renders', () => {
  const component = renderer.create(
    <Value
      value={1000}
      active={true}
      handleValueSelected={functionMock}
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
