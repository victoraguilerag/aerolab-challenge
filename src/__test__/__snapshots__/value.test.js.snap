// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value renders 1`] = `
<div
  className="active-value"
  style={
    Object {
      "background": "#0ad4fa",
      "borderRadius": "100px",
      "cursor": "pointer",
      "height": "40px",
      "margin": "0 5px",
    }
  }
>
  <div
    className="active-value-text"
    onClick={[Function]}
    style={
      Object {
        "alignItems": "center",
        "color": "#ffffff",
        "display": "flex",
        "fontSize": "1em",
        "height": "100%",
        "justifyContent": "center",
        "margin": "0 10px",
        "textAlign": "center",
      }
    }
  >
    1000
    <img
      alt="coin"
      className="coin"
      height="24px"
      src={Object {}}
      style={
        Object {
          "margin": "0 3px",
        }
      }
      width="24px"
    />
  </div>
</div>
`;
